paddle/fluid/operators/elementwise_op_function.h

```diff
@@ -381,15 +389,22 @@ static __global__ void ElemwiseGradBroadcast1CUDAKernel(
   } while (i < h);
 
   if (dy) {
+    T val = shm[threadIdx.x];
```

Comment:
There is no need to save too many variables in shared memory if we use `shfl_down`. `shfl_down` can access any variable, not just in shared_memory. It could be slightly faster.

Reply:
Done